### ISSUE TITLE
How to handle building Docker Images with protobuf cpp toolchain requirement on mac?

### ISSUE THREAD

**Issue by richwhitjr** [bazelbuild/rules_scala#851]
After recently upgrading the scala rules for our bazel build we noticed that we can no longer build docker images on mac for scala projects.  The problems seems to be that a recent change made it so the scala toolchain will build the protobuf library from source which requires the cpp toolchain.

Previously we would run `bazel run --cpu=k8 //:my_scala_docker_target` and build a docker image we could push. This now fails with:

```
ERROR: While resolving toolchains for target @io_bazel_rules_scala//src/java/io/bazel/rulesscala/scalac:scalac: no matching toolchains found for types @bazel_tools//tools/cpp:toolchain_type
ERROR: Analysis of target '//src/pipelines/examples/spark:bin' failed; build aborted: no matching toolchains found for types @bazel_tools//tools/cpp:toolchain_type
```

Anyone find a way to workaround this?

**Comment by ittaiz** [comment]
We're running with remote execution for a long time which require a lot of
toolchains/platform incantations so I think we're not feeling this.
On our linux machines we add
`--extra_toolchains=@bazel_toolchains//configs/ubuntu16_04_clang/latest:toolchain_default`
but it actually seems that on macs we're not passing it and I'm not sure
we're passing anything explicitly. On mac the default behavior AFAIR is to
use xcode. Maybe you're disabling it somehow?

**Comment by richwhitjr** [comment]
Yeah my situation is a bit different since my docker image has a jar plus some golang code to help deal with kubernetes.  So without the golang code I can target the normal cpu and build for mac and the jar will of course work fine once in the docker container.

For the golang code though I need to set `--cpu=k8` which tries to pool to the k8 toolchain.  This fine for the golang build and the scala part but protobuf can not be built with that cpu since it has C++ targets.

Thanks for the pointers on toolchains, I can look into that and see if I can get the C++ protobuf targets building correctly under mac targeting the k8 toolchain.
